include/zephyr/net/hostname.h

```diff
@@ -49,6 +53,18 @@ static inline const char *net_hostname_get(void)
 }
 #endif /* CONFIG_NET_HOSTNAME_ENABLE */
 
+/**
+ * @brief Set the device hostname
+ *
+ * @param hostname new hostname as char array.
+ * @param len Length of the hostname array.
+ *
+ * @return 0 if ok, <0 on error
+ */
+#if IS_ENABLED(CONFIG_NET_HOSTNAME_DYNAMIC)
```

Comment:
Please use `#if defined()` with preprocessor directives. Plus, it'd be good to define a stub function (returning an error) in case the feature is not enabled.